tensorflow/python/ops/array_ops.py

```diff
@@ -3503,7 +3503,7 @@ def sparse_placeholder(dtype, shape=None, name=None):
   # default shape out of the placeholder. Override that
   # shape to be the value determined here, so partial shapes can be
   # propagated.
-  result.set_shape(dense_shape_default)
+  result._dense_shape_default = dense_shape_default
```

Comment:
Is this a correct change? It changes functionality?

Reply:
It was refactored automatically by Pycharm IDE. I am not sure, so reverting back the changes to original. 
Thank you!